main.go

```diff
@@ -38,6 +39,33 @@ func main() {
 	defaultRegistry := flag.String("default-registry", "", fmt.Sprintf("default registry to use in absence of a fully qualified image name, defaults to %q", name.DefaultRegistry))
 	flag.Var(cp, "capath", "path to a file that contains CA certificates in the PEM format") // named after the curl cli flag
 
+	var forceCheckDisabledControllerKinds []string
+	flag.Func("force-check-disabled-controllers", `comma-separated list of controller kinds for which image is forcibly checked, even when workloads are disabled or suspended. Acceptable values include "Deployment", "StatefulSet", "DaemonSet", "Cronjob" or "*" for all kinds (this option is case-insensitive)`, func(flagValue string) error {
+		allowedControllerKinds := []string{"deployment", "statefulset", "daemonset", "cronjob"}
+
+	flagLoop:
+		for _, kind := range strings.Split(flagValue, ",") {
+			kind = strings.ToLower(kind)
+			if kind == "*" {
+				forceCheckDisabledControllerKinds = allowedControllerKinds
+				return nil
+			}
```

Comment:
This check should happen before the loop. Otherwise, the following flag values will be valid 
```
DaemonSet,*
StatefulSet,*,Deployment
```
which may be a little puzzling. The only valid values that should allow all kinds is `*`.

Reply:
✅